frontend/src/components/relation/delete-relation.tsx

```diff
@@ -0,0 +1,209 @@
+"use client";
+import { Relation } from "@/definitions";
+import {
+  Button,
+  Callout,
+  Dialog,
+  DialogBody,
+  DialogFooter,
+  Divider,
+  Icon,
+  IconName,
+  Tag,
+  Text,
+} from "@blueprintjs/core";
+import SquareIcon, { SquareIconSize } from "../icon/square-icon";
+import { ErrorDisplay } from "../error-display";
+import { useDeleteRelation } from "@/data/use-relations";
+import { mutate } from "swr";
+import { useTable } from "@/data/use-tables";
+import Loading from "@/app/loading";
+
+export default function DeleteRelation({
+  relation,
+  setRelation,
+}: {
+  relation: Relation | null;
+  setRelation: (relation: Relation | null) => void;
+}) {
+  const { isMutating, trigger: deleteRelation } = useDeleteRelation(
+    relation ? relation.id : undefined,
+  );
+  const {
+    data: table1,
+    isLoading: isLoadingTable1,
+    error: table1Error,
+  } = useTable(relation?.table_1);
+  const {
+    data: table2,
+    isLoading: isLoadingTable2,
+    error: table2Error,
+  } = useTable(relation?.table_2);
+  const {
+    data: joinTable,
+    isLoading: isLoadingJoinTable,
+    error: joinTableError,
+  } = useTable(relation?.join_table ?? undefined);
```

Comment:
Why explicitly undefined? and why ?? instead of ||? Prefer null, should avoid undefined.